src/app/tickets/ticket.service.ts

```diff
@@ -1,36 +1,92 @@
-    new Ticket(
-      'Trouble Accessing Site',
-      'Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perferendis esse nemo eligendi tempora, molestiae voluptatem quia et cum obcaecati dolor?',
-      'https://www.aarki.com/hs-fs/hubfs/blog%20header_deep%20thinking_2%20copy%20(2).jpg?width=2044&name=blog%20header_deep%20thinking_2%20copy%20(2).jpg',
-      [new KTool('ABC', 1), new KTool('EFG', 20)]
-    ),
-    new Ticket(
-      'Training Dates and Sign up',
-      'Lorem ipsum dolor sit amet consectetur adipisicing elit. Totam quisquam harum sit!',
-      'https://www.aarki.com/hs-fs/hubfs/blog%20header_deep%20thinking_2%20copy%20(2).jpg?width=2044&name=blog%20header_deep%20thinking_2%20copy%20(2).jpgE',
-      [new KTool('WWE', 2), new KTool('NBC', 1)]
-    ),
-  ];
-
-  constructor(private ktService: KToolsService) {}
-
-  getTickets() {
-    return this.tickets.slice();
-  }
-
-  getTicket(index: number) {
-    return this.tickets[index];
-  }
-
-  addToKTools(ingredients: KTool[]) {
-    this.ktService.addkTools(ingredients);
-  }
-}
+import { KToolsService } from '../k-tools/k-tools.service';
+import { Injectable } from '@angular/core';
+import { Ticket } from './ticket.model';
+// import { KTool } from '../shared/k-tool.model';
+import { Observable, of, Subject } from 'rxjs';
+
+@Injectable({ providedIn: 'root' })
+export class TicketService {
+  ticket$ = new Subject<Ticket[]>();
+
+  private tickets: Ticket[] = [
+    {
+      id: 'CBS-0001',
+      description: 'test one',
+      title: 'test one',
+      ticketNumber: 'C-1',
+      serviceDeskTypes: 'Report a bug',
+      kTool: 'Nationals',
+    },
+    {
+      id: 'CBS-0002',
+      description: 'test one',
+      title: 'test one',
+      ticketNumber: 'C-1',
+      serviceDeskTypes: 'Report a bug',
+      kTool: 'Astros',
+    },
+    {
+      id: 'CBS-0003',
+      description: 'test one',
+      title: 'test one',
+      ticketNumber: 'C-1',
+      serviceDeskTypes: 'Report a bug',
+      kTool: 'Astros',
+    },
+    {
+      id: 'CBS-0005',
+      description: 'test one',
+      title: 'test one',
+      ticketNumber: 'C-1',
+      serviceDeskTypes: 'Report a bug',
+      kTool: 'Astros',
+    },
+    {
+      id: 'CBS-0004',
+      description: 'test one',
+      title: 'test ondasdfe',
+      ticketNumber: 'C-1',
+      serviceDeskTypes: 'Report a bug',
+      kTool: 'Dodgers',
+    },
+    {
+      id: 'CBS-0006',
+      description: 'test one',
+      title: 'test ondasdfe',
+      ticketNumber: 'C-1',
+      serviceDeskTypes: 'Report a bug',
+      kTool: 'Nationals',
+    },
+  ];
+
+  constructor(private ktService: KToolsService) {}
+
+  getTickets(): Ticket[] {
+    return this.tickets.slice();
+  }
+
+  getTicket(ticketNo): Ticket {
+    const ticket = this.tickets.filter((x) => x.ticketNumber === ticketNo);
+    return ticket[0];
+  }
+
+  filterTickets(filteredValue): void {
+    this.tickets.filter = filteredValue.trim().toLowerCase();

```

Comment:
I think this would override the Array.prototype.filter method for the tickets array.